RecoEgamma/EgammaElectronProducers/src/LowPtGsfElectronSeedHeavyObjectCache.cc

```diff
@@ -0,0 +1,33 @@
+#include "RecoEgamma/EgammaElectronProducers/interface/LowPtGsfElectronSeedHeavyObjectCache.h"
+#include "TMVA/MethodBDT.h"
+#include "TMVA/Reader.h"
+#include <string>
+
+namespace lowpt {
+  HeavyObjectCache::HeavyObjectCache( const edm::ParameterSet& conf ) {
+    
+    edm::FileInPath weights( conf.getParameter<std::string>("Weights") );
+    
+    TMVA::Reader reader("!Color:Silent");
+    reader.AddVariable("trk_pt",&trk_pt);
+    reader.AddVariable("trk_eta",&trk_eta);
+    reader.AddVariable("trk_phi",&trk_phi);
+    reader.AddVariable("trk_p",&trk_p);
+    reader.AddVariable("trk_charge",&trk_charge);
+    reader.AddVariable("trk_nhits",&trk_nhits);
+    reader.AddVariable("trk_high_purity",&trk_high_purity);
+    reader.AddVariable("trk_inp",&trk_inp);
+    reader.AddVariable("trk_outp",&trk_outp);
+    reader.AddVariable("trk_chi2red",&trk_chi2red);
+    reader.AddVariable("preid_trk_ecal_Deta",&preid_trk_ecal_Deta);
+    reader.AddVariable("preid_trk_ecal_Dphi",&preid_trk_ecal_Dphi);
+    reader.AddVariable("preid_e_over_p",&preid_e_over_p);
+    reader.AddVariable("preid_gsf_dpt",&preid_gsf_dpt);
+    reader.AddVariable("preid_trk_gsf_chiratio",&preid_trk_gsf_chiratio);
+    reader.AddVariable("preid_gsf_chi2red",&preid_gsf_chi2red);
+    reader.BookMVA("BDT", weights.fullPath().c_str());
+    
+    gbr.reset( new GBRForest(dynamic_cast<TMVA::MethodBDT*>(reader.FindMVA("BDT"))) );
```

Comment:
Yeah, I've already seen the changes in 10_2 and I'll have to migrate ... And yes, indeed, black magic ...

Reply:
Yep, creating the GBRForests is a one-liner now, since 9.4.X in fact, so you can easily switch:

https://github.com/cms-sw/cmssw/blob/CMSSW_9_4_X/RecoEgamma/EgammaTools/interface/GBRForestTools.h

In 10.4.X, it moved to the new CommonTools/MVAUtils package Mauro was pointing to. The new implementation without the TMVA reader cuts loading time in half so it's worth moving to the GBRForestTools.

Reply:
Hi @guitargeek - is the latest interface available in 9_4_10? Or do I need to use a merge-topic? 

Also, can you point me to an example in 10_4_X? Do I need to explicitly list the input variables used?

Reply:
Yes, the latest interface is already available in 9_4_10, the header file is just at a different location. Here an example of a MVA heavy object cache how it should be done now:

https://github.com/cms-sw/cmssw/blob/master/RecoParticleFlow/PFProducer/interface/PFEGammaHeavyObjectCache.h

If you'd replace...
```#include "CommonTools/MVAUtils/interface/GBRForestTools.h"```
...with...
```#include "RecoEgamma/EgammaTools/interface/GBRForestTools.h"```
...this pattern should work in 9_4_X since 9_4_5.